tests/unit/lms/resources/_js_config/__init___test.py

```diff
@@ -309,6 +302,24 @@ def test_it_adds_the_viaUrl_api_config(self, url, via_url, js_config):
 
         assert js_config.asdict()["api"]["viaUrl"] == via_url
 
+    @pytest.mark.parametrize("is_admin", (True, False))
+    def test_canvas_studio_adds_config(
+        self, js_config, canvas_studio_service, is_admin
+    ):
+        canvas_studio_service.is_admin.return_value = is_admin
+
+        document_url = "canvas-studio://media/media_id"
+        js_config.add_document_url(document_url)
+
+        expected_auth_url = None
+        if is_admin:
+            expected_auth_url = "http://example.com/api/canvas_studio/oauth/authorize"
+
+        assert js_config.asdict()["api"]["viaUrl"] == {
+            "authUrl": expected_auth_url,
+            "path": "/api/canvas_studio/via_url",
+        }
```

Comment:
I'd probably split this into two separate, simpler tests, rather than one test with an `if`. Placing a high value on keeping tests dead simple even if it creates a little duplication (or I'd use fixtures to remove the setup duplication). The second test only needs to assert the value of `js_config.asdict()["api"]["viaUrl"]["authUrl"]`, not the rest of the object.